### DISCUSSION TITLE
How can I draw a horizontal line right next to some text with relative positioning?

### DISCUSSION THREAD

**Discussion by anantajit** [typst/typst#4727] [Q&A] [answered]
I'm trying to draw a line next to text for a "fill in the blank" type of template. 

```
Text
#h(1cm) // Inserting this or removing it had no effect
#line(length: 20%)
```

However, this still places the horizontal line on the next line on the page. Would appreciate any insights on why this won't work, and how I can fix it! This seems like a relatively common task, yet I wasn't able to find any documentation which explained how to achieve this.

**Comment by anantajit** [comment, marked as answer]
I believe the solution is to "box" the line:

```
text: #box(height: 14pt)[#line(length: 20%)]
```

**Reply by Mc-Zen** [reply]
Indeed, most things are actually block-level. Putting them into a `box` gives the desired effect :)
